docs/MANUAL-TESTING.md

```diff
@@ -0,0 +1,114 @@
+Manual Testing
+==============
+
+This document describes a high-level script of manual tests to check for. We
+should aim to replace items on this list with automated Spectron test cases.
+
+Image Selection
+---------------
+
+- [ ] Cancel image selection dialog
+- [ ] Select an unbootable image (without a partition table), and expect a
+  sensible warning
+- [ ] Attempt to select an unsupported image
+- [ ] Change image selection
+- [ ] Select a Windows image, and expect a sensible warning
+
+Drive Selection
+---------------
+
+- [ ] Open the drive selection modal
+- [ ] Remove all drives
+- [ ] Switch drive selection
+- [ ] Insert a single drive, and expect auto-selection
+- [ ] Insert more than one drive, and don't expect auto-selection
+- [ ] Insert a locked SD Card and expect a warning
+- [ ] Insert a too small drive and expect a warning
+- [ ] Put an image into a removable drive and attempt to flash the image to the
+  drive that contains it
+- [ ] Attempt to flash a compressed image into a drive that is big enough to
+  hold the compressed image, but not big enough to hold the uncompressed
+  version
+- [ ] Enable "Unsafe Mode" and attempt to select a system drive
+- [ ] Enable "Unsafe Mode", and if there is only one system drive (and no
+  removable ones), don't expect autoselection
+
+Image Support
+-------------
+
+Run the following tests with and without validation enabled:
+
+- [ ] Flash an uncompressed image
+- [ ] Flash a Bzip2 image
+- [ ] Flash a XZ image
+- [ ] Flash a ZIP image
+- [ ] Flash a GZ image
+- [ ] Flash a DMG image
+- [ ] Flash an image whose size is not a multiple of 512 bytes
+- [ ] Flash a compressed image whose size is not a multiple of 512 bytes
+- [ ] Flash an archive whose image size is not a multiple of 512 bytes
+- [ ] Flash an archive image containing a logo
+- [ ] Flash an archive image containing a blockmap file
+- [ ] Flash an archive image containing a manifest metadata file
+
+Flashing Process
+----------------
+
+- [ ] Disable validation and flash an image
+- [ ] Unplug the drive during flash or validation
```

Comment:
I was going to suggest we could also add the unplug drive during the elevation dialog test, but it might not be so relevant once the `child-writer` re-architecture goes through

Reply:
Might make sense to add it anyway - we can always remove it later.

Reply:
Done